packages/block-editor/src/components/colors/use-colors.js

```diff
@@ -219,26 +221,10 @@ export default function __experimentalUseColors(
 		return (
 			( needsBackgroundColor || needsColor ) &&
 			withFallbackStyles(
```

Comment:
This won't work, `withFallbackStyles` returns a component and calls the passed-in function with its root node's ref upon mounting.

Now we can query the new target ref here.

Reply:
I know, I was still figuring out that part. Rewrote it now. Looks like we need to use state to set the colours so the component rerenders?

Reply:
That's what this dep is for:

https://github.com/WordPress/gutenberg/blob/a7860eecc11878403b33ccf2dcfa16672179bad6/packages/block-editor/src/components/colors/use-colors.js#L248-L254

Reply:
I meant for `<ColorPanel>` to rerender. How does it know the detected colours if it's a ref?

Reply:
The detected color can't change without you doing something that updates the component and, therefore, the color panel.

There were some edge cases, but we decided they were reasonable.